### DISCUSSION TITLE
Next.js 13 eCommerce dynamic routing

### DISCUSSION THREAD

**Discussion by SalmenBejaoui** [vercel/next.js#48597] [Help]
### Summary

My current eCommerce has dynamic routes. the URLs have the following schema:

Product List Page: /:locale/:categoryId/:subCategoryId
Example :
```
/en-us/category-a
/en-ca/category-b
/en-uk/category-a/category-a-a
```

Product Details Page /:locale/:categoryId/:productCode
Example:
```
/en-us/category-a/abc-1234567788
/en-ca/category-b/abcdesf-121122113
/fr-fr/category-c/abcabc-1111111
```

The categories and products name are all dynamic. The only difference is that the product page URL finishes with "-{numer}"
How can I structure my Next.js project to match all URLs?

**Comment by stellar-pxl** [comment]
Have you looked at the appropriate section of the documentation?
`https://nextjs.org/docs/routing/dynamic-routes#catch-all-routes`
in your case: locale/[...slug].tsx - will match/category-a/abc-1234567788 or /category-b/abcdesf-121122113
Then you get this in the router object from the useRouter() hook.
